src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,56 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Field;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private final Map<Class<?>, Class<?>> interfaceImplementations;
+
+    private Injector() {
+        interfaceImplementations = Map.of(
+                FileReaderService.class, FileReaderServiceImpl.class,
+                ProductParser.class, ProductParserImpl.class,
+                ProductService.class, ProductServiceImpl.class
+        );
+    }
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Class<?> implClass = interfaceImplementations.get(interfaceClazz);
+
+        if (implClass == null || !implClass.isAnnotationPresent(Component.class)) {
+            throw new RuntimeException(
+                    "Injection failed, missing @Component annotation on the class "
+                    + implClass);
+        }
+
+        try {
+            Object instance = implClass.getDeclaredConstructor().newInstance();
+
+            // Inject dependencies
```

Comment:
According to the checklist, you should check the instances map before creating a new instance to avoid creating multiple instances of singleton classes. Currently, the `getInstance` method creates a new instance every time it is called.